cmd/vcert/result_writer.go

```diff
@@ -173,24 +173,33 @@ func (o *Output) AsJKS(c *Config) ([]byte, error) {
 		},
 	}
 
-	var jksPassword []byte
+	var keyPass []byte
 
-	//setting as jksPassword the value in --jks-password (if it was provided) or the value from keyPassword( --key-password or pass phrase value)
+	//setting as keyPass the value from keyPassword(--key-password or pass phrase value) or the value in --jks-password
+	if c.KeyPassword != "" {
+		keyPass = []byte(c.KeyPassword)
+	} else {
+		keyPass = []byte(c.JKSPassword)
```

Comment:
We always want `keyPass = []byte(c.KeyPassword)`, never `keyPass = []byte(c.JKSPassword)` for consistency with other formats (i.e. in VCert the `--key-password` is always focused on protecting the private key).

Reply:
mmmm… but according the chat that we had in the related email, we agreed that when it isn’t provided the private key password, that is when the --no-prompt flag is provided and the -key-password is not provided, then we can use the -jks-password value for both, the store password and also for the key entry password, and that code is for that case… or am I missing something?

Reply:
Yeah, I don't want the value of the `--jks-password` ever being used for the _PrivateKeyEntry_, only for the _Store_.   The `--jks-password` is the optional and when not specified the value of the `--key-password` should be used for the _Store_ which is how that is currently coded.

Reply:
I understand the point that you mention... the private key entry should be protected only by the key password( or the pass phrase if that was provided by prompt action) and the -jks-password should be used only to provide the password for the key store, but as we commented, what about of the case when neither the -key-password and the pass phrase were not provided( --no-prompt)? The password for the key entry is mandatory, at least it is a limit given the keystore go library which we are using

Reply:
Yes, that's right.  If a user specifies `--no-prompt` without also specifying `--key-password` with `--format jks` the expected result is an error stating passwords must be at least 6 characters long (because the default key password value is the empty string).

Reply:
got it, in that way the user is required to provide a password for the key entry... I will do the change :-)

Reply:
Thank you @marcos-albornoz 😃

Reply:
done!